src/webgpu/api/validation/capability_checks/limits/maxColorAttachmentBytesPerSample.spec.ts

```diff
@@ -0,0 +1,188 @@
+  .params(kLimitBaseParams.combine('maxColorAttachmentsLimitMode', kLimitModes))
+  .fn(async t => {
+    const { limitTest, testValueName, maxColorAttachmentsLimitMode } = t.params;
+    await t.testDeviceWithRequestedLimits(
+      limitTest,
+      testValueName,
+      async ({ device, testValue, actualLimit, shouldError }) => {
+        const pipelineDescriptor = getPipelineDescriptor(device, actualLimit, testValue);
+        if (!pipelineDescriptor) {
+          return;
+        }
+
+        await t.expectValidationError(() => {
+          device.createRenderPipeline(pipelineDescriptor);
+        }, shouldError);
+      },
+      { maxColorAttachments: maxColorAttachmentsLimitMode }
+    );
+  });
+
+g.test('createRenderPipelineAsync,at_over')
+  .desc(`Test using at and over ${limit} limit in createRenderPipelineAsync`)
+  .params(kLimitBaseParams.combine('maxColorAttachmentsLimitMode', kLimitModes))
+  .fn(async t => {
+    const { limitTest, testValueName, maxColorAttachmentsLimitMode } = t.params;
+    await t.testDeviceWithRequestedLimits(
+      limitTest,
+      testValueName,
+      async ({ device, testValue, actualLimit, shouldError }) => {
+        const pipelineDescriptor = getPipelineDescriptor(device, actualLimit, testValue);
+        if (!pipelineDescriptor) {
+          return;
+        }
+
+        await t.shouldRejectConditionally(
+          'OperationError',
+          device.createRenderPipelineAsync(pipelineDescriptor),
+          shouldError
+        );
+      },
+      { maxColorAttachments: maxColorAttachmentsLimitMode }
+    );
+  });
+
+g.test('beginRenderPass,at_over')
+  .desc(`Test using at and over ${limit} limit in beginRenderPass`)
+  .params(kLimitBaseParams.combine('maxColorAttachmentsLimitMode', kLimitModes))
+  .fn(async t => {
+    const { limitTest, testValueName, maxColorAttachmentsLimitMode } = t.params;
+    await t.testDeviceWithRequestedLimits(
+      limitTest,
+      testValueName,
+      async ({ device, testValue, actualLimit, shouldError }) => {
+        const numRGBA32F = actualLimit / 16;
+        const numR8Unorm = testValue - actualLimit;
+        const maxColorAttachments = device.limits.maxColorAttachments;
+        if (numRGBA32F + numR8Unorm > maxColorAttachments) {
+          return;
+        }
+
+        const encoder = device.createCommandEncoder();
+
+        const textures = [
+          ...createTextures(device, 'rgba32float', numRGBA32F),
+          ...createTextures(device, 'r8unorm', numR8Unorm),
+        ];
+
+        const pass = encoder.beginRenderPass({
+          colorAttachments: range(testValue, i => ({
+            view: textures[i].createView(),
+            loadOp: 'clear',
+            storeOp: 'store',
+          })),
+        });
+        pass.end();
+
+        await t.expectValidationError(() => {
+          encoder.finish();
+        }, shouldError);
+
+        textures.forEach(texture => texture.destroy());
```

Comment:
I think this is what `t.trackForCleanup` is supposed to be used for - instead of this forEach

Reply:
fixed.